### PR TITLE
Ssh keys (refactoring and tests)

### PR DESCRIPTION
**Pull request by warpc**
- refactoring mcollective ssh keys part;
- refactoring astute ssh keys part;
- add tests for astute ssh keys part.

### CODE REVIEW

File: spec/unit/deployment_engine_spec.rb [hunk 1]
```diff
@@ -98,9 +98,112 @@ class Engine < Astute::DeploymentEngine; end
...
+      
+      it 'should run ssh key generation with correct command' do
+        FileUtils.stubs(:mkdir_p).returns(true)
+        key_path = File.join(Engine::KEY_DIR, nodes.first['deployment_id'].to_s, 'nova', 'nova')
+        Engine.any_instance.expects(:system).with("ssh-keygen -b 2048 -t rsa -N '' -f #{key_path}").returns(true)
+        
+        deployer.deploy(nodes)
+      end
+      
+      it 'should not overwrite files' do
```
Comment (yoojinl) [inline spec/unit/deployment_engine_spec.rb:165]: I think will be cool to add case

```
it 'should overwrite files if key set to true' do 
```

**Comment by yoojinl** [comment]
++

File: spec/unit/deployment_engine_spec.rb (hunk 1 above)
Comment (yoojinl) [inline spec/unit/deployment_engine_spec.rb:165]: Ok I've got it

File: lib/astute/deployment_engine.rb [hunk 2]
```diff
@@ -89,17 +88,11 @@ def include_node?(nodes_array, node)
     end
     
     # Generate and upload ssh keys from master node to all cluster nodes.
-    def generate_and_upload_ssh_keys(key_names, node_uids, deployment_id)
-      upload_mclient = MClient.new(@ctx, "uploadfile", node_uids)
-      
-      key_names.each do |key_name|
+    def generate_and_upload_ssh_keys(node_uids, deployment_id)
+      raise "Deployment_id is missing" unless deployment_id
+      Astute.config.PUPPET_SSH_KEYS.each do |key_name|
```
Comment (aglarendil) [inline lib/astute/deployment_engine.rb:93]: I do not see any filtering by nodes. E.g., nodes not hosting ceph should not have ceph installed. Is this planned?

**Comment by aglarendil** [comment]
+1

**Comment by warpc** [comment]
success test — fuellib_pull.iso#9.

**Pull request merged**